cpg_infra/driver.py

```diff
@@ -727,6 +728,16 @@ def setup_web_access_group_memberships(self):
             member=self.analysis_group,
         )
 
+    def setup_service_account_generation_permission(self):
+        if not isinstance(self.infra, GcpInfrastructure):
+            return
+
+        self.infra.add_project_role(
```

Comment:
Can we add this on the service account itself instead of the whole project?

Reply:
Yep, definitely makes sense